### ISSUE TITLE
Document Edited indicator on iTerm

### ISSUE THREAD

**Issue by RSickenberg** [withfig/fig#718]
### Description:
> Please include a detailed description of the issue (and an image or screen recording, if applicable)
Hello, my issue is quite simple, the dot in the red button to close has the document edited indicator. 
Since I only have an ZSH session, the `figterm` attached shouldn't trigger it :)
I can override it but it will lose its importance.

Regards 😄 

![image](https://user-images.githubusercontent.com/16922548/146346065-320536a6-3376-4367-bcb0-40eda3bd9587.png)

### Details:
|macOS|Fig|Shell|
|-|-|-|
|12.1.21C52|1.0.53|/bin/zsh|
<details><summary><code>fig diagnostic</code></summary>
<p>
<pre>Fig Version: Version 1.0.53 (B339) [Swiss French] 
UserShell: /bin/zsh
Bundle path: /Applications/Fig.app
Autocomplete: true
Settings.json: true
CLI installed: true
CLI tool path: /usr/local/bin/fig
Accessibility: true
Number of specs: 0
SSH Integration: false
Tmux Integration: false
Keybindings path: <none>
iTerm Integration: installed!
Hyper Integration: application is not present.
VSCode Integration: installed!
Docker Integration: false
Symlinked dotfiles: false
Only insert on tab: false
Installation Script: true
PseudoTerminal Path: 
SecureKeyboardInput: false
SecureKeyboardProcess: <none>
Current active process: /bin/zsh (41842) - /dev/ttys001
Current working directory: /Users/romainsickenberg
Current window identifier: 803/% (com.googlecode.iterm2)
Path: /Users/romainsickenberg/.nvm/versions/node/v14.18.0/bin:/opt/homebrew/bin:/usr/local/sbin:/Users/romainsickenberg/.zplug/repos/zplug/zplug/bin:/Users/romainsickenberg/.zplug/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/MacGPG2/bin:/usr/local/share/dotnet:~/.dotnet/tools:/Library/Apple/usr/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Applications/Wireshark.app/Contents/MacOS:/Users/romainsickenberg/.fig/bin:/Users/romainsickenberg/.rvm/bin
Fig environment variables:
  - TERM_SESSION_ID=w0t0p0:AE64238C-070B-4B7F-A3D3-E56A5EADB1BB
  - FIG_INTEGRATION_VERSION=5
  - FIG_TERM=1
  - FIG_TERM_VERSION=3
  - FIG_ENV_VAR=1
  - FIG_CHECKED_PROMPTS=1
</pre>
</p>
</details>

**Comment by Zakini** [comment]
I'm also getting this

| macOS  | Fig    | Shell | Terminal |
|--------|--------|-------|----------|
| 12.0.1 | 1.0.53 | zsh   | iTerm    |

I don't know much about how Fig works, but it looks like it adds another version of your shell. In Activity Monitor, I can see `zsh` running _and_ `zsh (figterm)` running.

I found a workaround for this though; since it's not 'regular' zsh that's running, you need to tell iTerm to ignore the fig version of zsh when it decides whether to prompt you on close:
![Screenshot 2021-12-16 at 12 06 51](https://user-images.githubusercontent.com/11508697/146369125-006e703a-9819-4df3-bb4a-e386f2d4445c.png)

Is this a bug, or is this just something that needs mentioning during onboarding or in the docs somewhere?

**Comment by mschrage** [comment]
Hey @RSickenberg and @Zakini! Thanks for reporting this. 

I'm not too familiar with how the Document Edited indicator normally works. Can you describe what the expected behavior should be?

**Comment by Zakini** [comment]
@mschrage Basically it means that iTerm won't let you close the window/tab without confirming first with a popup like this:
![Screenshot 2021-12-16 at 19 57 35](https://user-images.githubusercontent.com/11508697/146440642-84fa477b-594c-4e64-bee9-7ebac54950eb.png)

It's useful when you've got something running still, like if you're part through editing something in nano like in the screenshot above, but it shouldn't happen if nothing's running.

This is what I get (without the workaround I mentioned earlier):
![Screenshot 2021-12-16 at 20 01 52](https://user-images.githubusercontent.com/11508697/146441166-741a7e9f-b36a-4e15-942f-11dc2cd83445.png)

**Comment by mschrage** [comment]
Okay I see now. Interestingly, I don't get this popup alert at all when I close an iTerm session. Did you have to change some setting in order for this to show up originally? @Zakini

**Comment by Zakini** [comment]
Even if you've got something running in the window when you try to close it @mschrage?

I don't know what the default settings are for iTerm (I set it up on my machine ages ago). Try comparing my settings in the screenshot in my first comment with yours. If I had to guess, I assume you have "Prompt before closing?" set to "Never".

**Comment by mschrage** [comment]
> If I had to guess, I assume you have "Prompt before closing?" set to "Never".

You're right. I do have "Prompt before closing?" set to "Never". (I think this is the default). 

If adding `zsh` to the list solves the issue, for the time being, we can send people to this thread for the workaround. 

I'm not sure what heuristics iTerm is using to decide when to turn on the Document Edited indicator. I'll try to investigate this at some point.
